From 573339c5e31320f50e80b9800fb1de0af26bf001 Mon Sep 17 00:00:00 2001
From: Leandro Pereira <leandro.pereira@intel.com>
Date: Tue, 23 Apr 2019 10:17:19 -0700
Subject: [PATCH] Ignore -Werror if ${GCC_IGNORE_WERROR} environment variable
 is set

-Werror should be set only by developers, not consumers of a project's
source code.  By providing this environment variable, the package build
system can signal that -Werror should never be used without the need to
patch each individual package build system.
---
 gcc/opts.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gcc/opts.c b/gcc/opts.c
index 02f6b4656..1bd011406 100644
--- a/gcc/opts.c
+++ b/gcc/opts.c
@@ -2321,7 +2321,10 @@ common_handle_option (struct gcc_options *opts,
       break;
 
     case OPT_Werror:
-      dc->warning_as_error_requested = value;
+      if (getenv("GCC_IGNORE_WERROR"))
+        dc->warning_as_error_requested = false;
+      else
+        dc->warning_as_error_requested = value;
       break;
 
     case OPT_Werror_:
-- 
2.21.0

